### ISSUE TITLE
Missing property 'project' creating a js library with rollup and swc

### ISSUE THREAD

**Issue by davidmendezph** [nrwl/nx#12843]
## Current Behavior

When running the following command to generate a new library using rollup and swc
```sh
nx generate @nrwl/js:library packagename \
--bundler=rollup \
--compiler=swc \
--importPath=@company/packagename \
--publishable \
--setParserOptionsProject \
--testEnvironment=node
```
the project generated succesfully but when running `nx build packagename` it fails with the message `Required property 'project' is missing`. `--verbose` returns no more information either.
 
## Expected Behavior

The library should generate and the build command should not fail.

### Environment

`Dependencies`
```json
  "devDependencies": {
    "@babel/core": "7.12.13",
    "@babel/preset-typescript": "7.12.13",
    "@nrwl/cli": "14.8.3",
    "@nrwl/cypress": "14.8.3",
    "@nrwl/eslint-plugin-nx": "14.8.3",
    "@nrwl/jest": "14.8.3",
    "@nrwl/linter": "14.8.3",
    "@nrwl/react": "14.8.3",
    "@nrwl/rollup": "14.8.3",
    "@nrwl/storybook": "14.8.3",
    "@nrwl/web": "14.8.3",
    "@nrwl/workspace": "14.8.3",
    "@semantic-release/exec": "^6.0.3",
    "@semantic-release/git": "^10.0.1",
    "@storybook/addon-essentials": "~6.5.9",
    "@storybook/builder-webpack5": "~6.5.9",
    "@storybook/core-server": "~6.5.9",
    "@storybook/manager-webpack5": "~6.5.9",
    "@storybook/react": "~6.5.9",
    "@storybook/test-runner": "^0.7.2",
    "@svgr/webpack": "^5.4.0",
    "@swc/cli": "~0.1.55",
    "@swc/core": "^1.2.173",
    "@swc/jest": "0.2.20",
    "@testing-library/react": "13.4.0",
    "@types/color": "^3.0.3",
    "@types/jest": "28.1.1",
    "@types/node": "16.11.7",
    "@types/react": "18.0.20",
    "@types/react-dom": "18.0.6",
    "@typescript-eslint/eslint-plugin": "^5.39.0",
    "@typescript-eslint/parser": "^5.39.0",
    "babel-jest": "28.1.1",
    "babel-loader": "8.1.0",
    "css-loader": "^6.7.1",
    "cypress": "^10.7.0",
    "eslint": "~8.15.0",
    "eslint-config-airbnb": "^19.0.4",
    "eslint-config-airbnb-typescript": "^17.0.0",
    "eslint-config-prettier": "^8.1.0",
    "eslint-plugin-cypress": "^2.10.3",
    "eslint-plugin-import": "^2.26.0",
    "eslint-plugin-jsx": "^0.1.0",
    "eslint-plugin-jsx-a11y": "^6.6.1",
    "eslint-plugin-prettier": "^4.2.1",
    "eslint-plugin-react": "^7.31.8",
    "eslint-plugin-react-hooks": "^4.6.0",
    "jest": "28.1.1",
    "jest-environment-jsdom": "28.1.1",
    "last-release-git": "^0.0.3",
    "nx": "14.8.3",
    "prettier": "2.7.1",
    "react-test-renderer": "18.2.0",
    "semantic-release": "^19.0.5",
    "ts-jest": "28.0.5",
    "ts-node": "10.9.1",
    "typescript": "4.8.4",
    "url-loader": "^3.0.0"
},
  "dependencies": {
    "@emotion/react": "11.10.4",
    "@emotion/styled": "11.10.4",
    "@swc/helpers": "~0.3.3",
    "color": "4.2.3",
    "core-js": "^3.6.5",
    "prop-types": "^15.8.1",
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "regenerator-runtime": "0.13.7",
    "tslib": "^2.3.0"
  }
```

## Possible fix
If you add the property `project`, the build succeeds
```json
"options": {
        "outputPath": "dist/packages/packagename",
        "main": "packages/packagename/src/index.ts",
        "tsConfig": "packages/packagename/tsconfig.lib.json",
        "compiler": "swc",
        "project": "packages/packagename/tsconfig.lib.json", <--- this property
        "assets": []
      }
```

## TLDR
The `--publishable` arg is not ensuring proper publishable library is generated.

## Nx 15.0.3
Tried migrating nx to version 15.0.3 but the error persists.

## Edit 11/2/2022
Had some issues publishing, was due to the project property not being the package.json.
The correct value should be:
```
 "project": "packages/packagename/package.json",
```

**Comment by ajwootto** [comment]
Also just ran into this on nx 15.0.9

**Comment by jacobtipp** [comment]
Had this happen to me with 15.5.1.

**Comment by mandarini** [comment]
Hello all! I can no longer reproduce this in latest version, I am pretty sure it was fixed. I will close this issue, but feel free to ping me and I will check it again if you retry with latest and the issue persists!
